### PR TITLE
fix: ensure promise tests are correct

### PR DESCRIPTION
**Pull request by dignifiedquire**
This fixes the test failures in ipfs/js-ipfs#392

@diasdavid please review

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
     "readable-stream": "1.1.13"
   },
   "devDependencies": {
-    "aegir": "^3.2.0"
+    "aegir": "^4.0.0"
```
Comment (diasdavid) [inline package.json]: wanna bump it to 5?

**Comment by diasdavid** [comment]
Could you give more info of what was the failure?

**Comment by diasdavid** [comment]
Have you tested it against js-ipfs-api?

**Comment by dignifiedquire** [comment]
This is the failure: (build: https://circleci.com/gh/ipfs/js-ipfs/665)

> 1) core --both .files promise API .add buffer:
     Error: Resolution method is overspecified. Specify a callback *or* return a Promise; not both.
    at webpack:///~/interface-ipfs-core/lib/files.js:226:0 <- test/browser.js:162249:13

  

  2) core --both .files promise API .add buffer:
     Resolution method is overspecified. Specify a callback *or* return a Promise; not both.

File: package.json (hunk 1 above)
Comment (dignifiedquire) [inline package.json]: :D sure

**Comment by diasdavid** [comment]
I'm surprised that this error just happened now. Can you tell what got broken meanwhile?

Otherwise LGTM

**Comment by dignifiedquire** [comment]
The error is part of `mocha@3` which checks for this explicitly, before it just go silently swallowed. The confusing part is that I actually didn't yet upgrade that branch to mocha 3, but it needs fixing in any case.

**Comment by dignifiedquire** [comment]
> Have you tested it against js-ipfs-api?

just tested and things pass

**Comment by diasdavid** [comment]
> mocha@3

Thank you for clarifying :)

**Pull request merged**